Tests/Foundation/Tests/TestFileManager.swift

```diff
@@ -275,16 +277,21 @@ class TestFileManager : XCTestCase {
 
             // test unExecutable if file has no permissions
             try fm.setAttributes([.posixPermissions : NSNumber(value: Int16(0o0000))], ofItemAtPath: path)
-#if os(Windows)
-            // Files are always executable on Windows
-            XCTAssertTrue(fm.isExecutableFile(atPath: path))
-#else
             XCTAssertFalse(fm.isExecutableFile(atPath: path))
-#endif
 
+#if os(Windows)
+            // test unExecutable even if file has an `exe` extension
+            try fm.copyItem(atPath: path, toPath: exePath)
+            XCTAssertFalse(fm.isExecutableFile(atPath: exePath))
+#else
```

Comment:
This test could still be run on Linux where it would also pass so I would suggest removing the `#if os(Windows)` and change the `#else` to `#if !os(Windows)` so this test is run on all platforms.

Reply:
This could run, but it’s meaningless. The two different tests actually reflect different behavior on POSIX and Windows systems, which I think should be parallel.

Reply:
Ideally we want to run as many of the tests across all platforms and reduce the number of platform specific tests. In this case I would suggest adding the `.exe` extension to the original filename, eliminate the copy and then the 0 byte `.exe` file should then pass as non-executable on all platforms.

Reply:
I still think merging these tests would lead to confusion, and the current tests can reflect actual cases... It's not about "running as many tests across platforms". Testing with ".exe" extension makes no sense on Linux and macOS, nor does testing POSIX permissions on Windows.

Reply:
I would simply add a comment above the file saying that it is intended to be seen as an not executable by windows even  though it has a `.exe` extension since it lacks the appropriate header, and not executable by unix since it lacks the appropriate execute bit.